# New York city deaths
**Basic EDA**
1. show shape of all the data and each column
2. deal missing or b=null values  
3. Describe 

In [61]:
import pandas as pd
import numpy as np
from math import *
import plotly.express as px

In [27]:
# Read the Dataset
deaths = pd.read_csv("./New_York_City_Leading_Causes_of_Death.csv")

In [21]:
deaths.head(5)

Year                                      Leading Cause Sex  \
0  2011       Chronic Lower Respiratory Diseases (J40-J47)   M   
1  2008              Malignant Neoplasms (Cancer: C00-C97)   M   
2  2010     Chronic Liver Disease and Cirrhosis (K70, K73)   M   
3  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
4  2014  Certain Conditions originating in the Perinata...   M   

       Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
0  White Non-Hispanic    444       33.3                    27.8  
1  White Non-Hispanic   3361      250.4                   214.3  
2            Hispanic    150       13.1                    17.6  
3            Hispanic   1146         96                   143.5  
4  Not Stated/Unknown     25          .                       .

In [22]:
deaths.tail(5)

Year                                      Leading Cause Sex  \
1089  2011  Mental and Behavioral Disorders due to Acciden...   M   
1090  2013            Influenza (Flu) and Pneumonia (J09-J18)   M   
1091  2007                        Diabetes Mellitus (E10-E14)   F   
1092  2010   Insitu or Benign / Uncertain Neoplasms (D00-D48)   F   
1093  2012                        Diabetes Mellitus (E10-E14)   M   

             Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
1089               Hispanic    145       12.6                      13  
1090               Hispanic    233       19.7                    33.1  
1091     White Non-Hispanic    231       16.1                     9.8  
1092  Other Race/ Ethnicity      .          .                       .  
1093     White Non-Hispanic    292       21.8                    18.1

In [23]:
deaths.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1094 entries, 0 to 1093
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Year                     1094 non-null   int64 
 1   Leading Cause            1094 non-null   object
 2   Sex                      1094 non-null   object
 3   Race Ethnicity           1094 non-null   object
 4   Deaths                   1094 non-null   object
 5   Death Rate               1094 non-null   object
 6   Age Adjusted Death Rate  1094 non-null   object
dtypes: int64(1), object(6)
memory usage: 60.0+ KB


### Some columns needs processing :

* "Deaths" : needs casting from object to int 
* Same for "Death Rate " and Age Adjusted Death Rate
  _______________________
  But if we tried to cast type we would found that missing values are "."

  so let's change them first

In [24]:
deaths.shape


(1094, 7)

In [25]:
deaths["Leading Cause"].nunique()

26

#### Since number of unique causes values are 26 and the number of rows is total of 1094 ,so we can interpolate the values using mean valus


In [29]:
# cHange periods 
deaths["Deaths"] = [int(x)  if x != '.' else 0 for x in deaths["Deaths"]]
deaths["Death Rate"] = [float(x)  if x != '.' else 0 for x in deaths["Death Rate"]]
deaths["Age Adjusted Death Rate"] = [float(x)  if x != '.' else 0 for x in deaths["Age Adjusted Death Rate"]]

In [30]:
# IN deaths let's just replace with the mean 
mean_d = np.mean(deaths["Deaths"])
mean_r = np.mean(deaths["Death Rate"])
mean_a = np.mean(deaths["Age Adjusted Death Rate"])

# replace zeros with mean
deaths["Deaths"] = [int(x)  if x !=0 else mean_d for x in deaths["Deaths"]]
deaths["Death Rate"] = [float(x)  if x !=0 else mean_r for x in deaths["Death Rate"]]
deaths["Age Adjusted Death Rate"] = [float(x)  if x !=0 else mean_a for x in deaths["Age Adjusted Death Rate"]]

In [10]:
deaths["Deaths"] = [mean_d  if x == 0 else 0 for x in deaths["Deaths"]]
deaths["Death Rate"] = [mean_r  if x == 0 else 0 for x in deaths["Death Rate"]]
deaths["Age Adjusted Death Rate"] = [mean_a if x == 0 else x for x in deaths["Age Adjusted Death Rate"]]

In [31]:
deaths.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1094 entries, 0 to 1093
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Year                     1094 non-null   int64  
 1   Leading Cause            1094 non-null   object 
 2   Sex                      1094 non-null   object 
 3   Race Ethnicity           1094 non-null   object 
 4   Deaths                   1094 non-null   float64
 5   Death Rate               1094 non-null   float64
 6   Age Adjusted Death Rate  1094 non-null   float64
dtypes: float64(3), int64(1), object(3)
memory usage: 60.0+ KB


In [32]:
deaths.columns

Index(['Year', 'Leading Cause', 'Sex', 'Race Ethnicity', 'Deaths',
       'Death Rate', 'Age Adjusted Death Rate'],
      dtype='object')

In [34]:
# Lets do some data understanding

deaths.describe()

Year       Deaths   Death Rate  Age Adjusted Death Rate
count  1094.000000  1094.000000  1094.000000              1094.000000
mean   2010.477148   437.484782    46.786156                46.806684
std       2.293419   822.884325    62.202794                56.302261
min    2007.000000     5.000000     2.400000                 2.500000
25%    2008.000000    47.000000    15.800000                16.950000
50%    2010.000000   186.000000    34.583821                34.598995
75%    2012.000000   388.480804    34.583821                34.598995
max    2014.000000  7050.000000   491.400000               350.700000

In [47]:
# Years
print("Years are \nfrom :" , min(deaths.Year))
print("To :" , max(deaths.Year) , " Approx ", deaths.Year.nunique() , " years")
print("Total number of deaths in ",deaths.Year.nunique()," years is ", ceil(sum(deaths.Deaths)) )

Years are 
from : 2007
To : 2014  Approx  8  years
Total number of deaths in  8  years is  478609


## Deep dive
 **Deaths** All deaths 
* avg of deaths for each cause is around 50 and also close from the adjusted rate which is around 45

In [35]:
deaths.head(5)

Year                                      Leading Cause Sex  \
0  2011       Chronic Lower Respiratory Diseases (J40-J47)   M   
1  2008              Malignant Neoplasms (Cancer: C00-C97)   M   
2  2010     Chronic Liver Disease and Cirrhosis (K70, K73)   M   
3  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
4  2014  Certain Conditions originating in the Perinata...   M   

       Race Ethnicity  Deaths  Death Rate  Age Adjusted Death Rate  
0  White Non-Hispanic   444.0   33.300000                27.800000  
1  White Non-Hispanic  3361.0  250.400000               214.300000  
2            Hispanic   150.0   13.100000                17.600000  
3            Hispanic  1146.0   96.000000               143.500000  
4  Not Stated/Unknown    25.0   34.583821                34.598995

# Analysis

* which Year has the most deaths ?
* what are the top 5 leading causes of deaths ?
* Deaths are common among males or females ?

In [37]:
# top 5 causes 
deaths['Race Ethnicity'].unique()

array(['White Non-Hispanic', 'Hispanic', 'Not Stated/Unknown',
       'Asian and Pacific Islander', 'Black Non-Hispanic',
       'Other Race/ Ethnicity'], dtype=object)

In [67]:
deaths.Year.nunique()

8

In [58]:
# 1 top 5 leading causes between past and present
top =deaths.sort_values(by=['Deaths'],ascending=False).head(5)
top

Year                                   Leading Cause Sex  \
402   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
455   2008  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
181   2009  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
728   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
1047  2008  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   

          Race Ethnicity  Deaths  Death Rate  Age Adjusted Death Rate  
402   White Non-Hispanic  7050.0       491.4                    250.7  
455   White Non-Hispanic  6836.0       476.2                    245.5  
181   White Non-Hispanic  6297.0       438.3                    226.2  
728   White Non-Hispanic  5632.0       421.0                    350.7  
1047  White Non-Hispanic  5503.0       410.0                    342.3

In [62]:
# first visual 
px.bar(top, x="Year", y="Deaths", color="Sex", title="Deaths by Year").show()

#### * In 2009 all deaths were only from females , is that a disease or what keep males safe then ?
#### * 2007 Has the top number of deaths of total 7050
#### * Second most deaths was in 2008 and 2009 , our data is until year 2014 , seems that life got better and less deaths after 2009

In [70]:
# Since year 2007 has the most deaths let's chech the top leading causes then 
d = deaths[deaths['Year'] == 2007]
d =d.sort_values(by=['Deaths'],ascending=False).head(5)
d

Year                                   Leading Cause Sex  \
402   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   
728   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   M   
1000  2007           Malignant Neoplasms (Cancer: C00-C97)   F   
575   2007           Malignant Neoplasms (Cancer: C00-C97)   M   
223   2007  Diseases of Heart (I00-I09, I11, I13, I20-I51)   F   

          Race Ethnicity  Deaths  Death Rate  Age Adjusted Death Rate  
402   White Non-Hispanic  7050.0       491.4                    250.7  
728   White Non-Hispanic  5632.0       421.0                    350.7  
1000  White Non-Hispanic  3518.0       245.2                    167.4  
575   White Non-Hispanic  3356.0       250.9                    213.7  
223   Black Non-Hispanic  2722.0       258.6                    247.3

In [71]:
# second visual 
px.bar(d, x="Leading Cause", y="Deaths", color="Sex", title="Deaths for each leading cause").show()

#### Disease of the heart is the top leading cause for deaths in 2007

In [79]:
# Distribution of deaths 
px.histogram(deaths,x='Deaths')

### Most of diseases cause death around 300 people